### PR TITLE
Add documentation for ImageReader

### PR DESCRIPTION
**Pull request by rizwangilani**
Added documentation for ImageReader.

### CODE REVIEW

**Comment by rizwangilani** [comment]
Updated PR to fix a ref-link.

File: README.md [hunk 1]
```diff
@@ -111,6 +111,39 @@ for epoch in range(num_epochs):
     reader.reset() # Return to the beginning of dataset.
 ```
 
+### Reading Images as NumPy Arrays
+
+MLIO supports reading image files in jpeg, png, and recordio formats, using the [`ImageReader`](doc/python/data_reader.md#ImageReader) class. The code snippet below shows how you can quickly iterate through an image dataset in mini-batches.
```
Comment (cbalioglu) [inline README.md]: recordio is not an image format, but rather a way to find the boundaries of images in a stream of bytes. I would just say `MLIO supports reading image files in JPEG and PNG formats using ...`.

File: README.md [hunk 2]
```diff
@@ -111,6 +111,39 @@ for epoch in range(num_epochs):
...
+
+from mlio.integ.numpy import as_numpy
+
+file_dir = '/path/to/image_dir' # This can be a directory or a single file.
+dataset = mlio.list_files(file_dir, pattern='*.jpg') # This can be jpg, png, or rec, depending on the image format.
+
+# Setting general DataReader parameters.
+data_reader_params = mlio.DataReaderParams(dataset=dataset, batch_size=200)
+
+# Setting ImageReader parameters. Set img_frame to mlio.ImageFrame.RECORDIO for recordio datasets.
```
Comment (cbalioglu) [inline README.md]: recordio is not a wellknown encoding for people who have not used MXNet before. Saying `for MXNet RecordIO datasets.` would make it more clear.

File: README.md [hunk 3]
```diff
@@ -111,6 +111,39 @@ for epoch in range(num_epochs):
...
+image_reader_prm = mlio.ImageReaderParams(img_frame=mlio.ImageFrame.NONE, resize=500, image_dimensions=[3,166,190], to_rgb=1)
```
Comment (cbalioglu) [inline README.md]: Make sure that your Python lines are less than 80 characters.

File: doc/python/data_reader.md [hunk 4]
```diff
@@ -77,6 +80,16 @@ RecordIOProtobufReader(data_reader_params : DataReaderParams)
 
 - `data_reader_params`: See[`DataReaderParams`](#DataReaderParams).
 
+## ImageReader
+Represents a data reader for reading image datasets in jpeg, png, and recordio formats.
```
Comment (cbalioglu) [inline doc/python/data_reader.md]: Same advice as above.

File: doc/python/data_reader.md [hunk 5]
```diff
@@ -284,6 +314,14 @@ Specifies how a batch that contains erroneous data should be handled.
 | `SKIP`  | Skip the batch.                           |
 | `WARN`  | Skip the batch and log a warning message. |
 
+### ImageFrame
+Specifies what image frame to use for reading an image dataset.
+
+| Value       | Description                                        |
+|-------------|----------------------------------------------------|
+| `NONE`      | For reading raw image files in jpeg or png format. |
+| `RECORDIO`  | For reading image files in recordio format.        |
```
Comment (cbalioglu) [inline doc/python/data_reader.md]: Again saying MXNet RecordIO would be more clear.

**Review by cbalioglu — APPROVED** [review]
Just a few minor comments. Looks good to me.

**Pull request closed without merging**